master/master.go

```diff
@@ -47,6 +58,10 @@ func (m *Master) Run() {
 			// (controller runs in the background and manages the number of instances)
 			// call load balancer function to schedule the tasks
 			// move tasks from `newTasks` to `scheduledTasks`
+			if m.firstTaskIsValid() && m.slotsAvailable() {
+				ip := m.scheduleTask()
+				SendTask(m.newTasks[0], ip)
+			}
```

Comment:
This part will only work if tasks arrive lower at the rate the incoming messages to `m.scheduleChan`. Not sure how to solve it, we need to either find a good balance between the rate or schedule multiple tasks at once.